### PR TITLE
MEN-3506 Porpagate status: migration 1.7.0 and set force migration ve…

### PR DESCRIPTION
**Pull request by merlin-northern**
…rsion

Migration is added:
 * 1.7.0 version bump
 * for automate upgrades

propagate-inventory-statuses command now supports force-set-migration
 * allows to set the migration version in the db

Misc:
 * store StoreMigrationVersion added

ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
I am not very proud of StoreMigrationVersion but I had to add it for the force-set-migration sake.

File: cmd/commands.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
```
Comment (tranchitella) [inline cmd/commands.go:20]: Can you please move this import lower, in the "non std-lib" import group?

File: cmd/commands.go [hunk 2]
```diff
@@ -374,16 +377,32 @@ func tryPropagateStatusesInventoryForDb(db store.DataStore, c cinv.Client, dbnam
...
 		err = updateDevicesStatus(ctx, db, c, tenant, status, dryRun)
 		if err != nil {
 			l.Infof("Done with DB %s status=%s, but there were errors: %s.", dbname, status, err.Error())
+			errReturned = err
 		} else {
 			l.Infof("Done with DB %s status=%s", dbname, status)
 		}
 	}
+	if migrationVersion != "" && !dryRun {
+		if err != nil {
```
Comment (tranchitella) [inline cmd/commands.go:391]: if the last updateDeviceStatus returns nil as error, you'll miss this if.
Are you sure here you want to check err instead of errReturned?

File: cmd/commands_test.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ package cmd
 import (
 	"context"
 	"fmt"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
```
Comment (tranchitella) [inline cmd/commands_test.go:19]: Same here, please move the import down.

File: store/datastore.go [hunk 4]
```diff
@@ -17,6 +17,7 @@ package store
 import (
 	"context"
 	"errors"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
```
Comment (tranchitella) [inline store/datastore.go:20]: Same here.

File: store/mongo/migration_1_7_0.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+)
+
+type migration_1_7_0 struct {
+	ms  *DataStoreMongo
+	ctx context.Context
+}
+
+const (
+	migrationContextTimeouts = 360
+	devicesBatchSize         = 512
```
Comment (tranchitella) [inline store/mongo/migration_1_7_0.go:41]: I love these 2^ numbers :)

File: store/mongo/migration_1_7_0.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	"github.com/mendersoftware/deviceauth/model"
+)
+
+type migration_1_7_0 struct {
+	ms  *DataStoreMongo
+	ctx context.Context
+}
+
+const (
+	migrationContextTimeouts = 360
```
Comment (tranchitella) [inline store/mongo/migration_1_7_0.go:40]: 360 seconds to set up to 512 device statuses? Maybe we can lower it, I believe 60 seconds are already too much!

File: store/mongo/migration_1_7_0.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+			Name:       &indexAuthSet_IdentityDataSha256_PubKey,
+			Unique:     &_true,
+		},
+	}
+	cAuthSets := m.ms.client.Database(ctxstore.DbFromContext(m.ctx, DbName)).Collection(DbAuthSetColl)
+	authSetIndexes := cAuthSets.Indexes()
+	_, err := authSetIndexes.CreateOne(m.ctx, authSetUniqueIndex)
+	if err != nil {
+		return errors.Wrap(err, "failed to create index containing IdDataSha256 and PubKey on auth sets")
+	}
```
Comment (tranchitella) [inline store/mongo/migration_1_7_0.go:108]: I think I already asked you this question in a different PR: is this index related to this migration?

Comment (merlin-northern) [reply store/mongo/migration_1_7_0.go]: so sorry about that! I copied the same piece of code, with apparently the same comment of yours. thanks!

File: store/mongo/migration_1_7_0.go (hunk 6 above)
Comment (merlin-northern) [reply store/mongo/migration_1_7_0.go]: sure.-> 32 the way you like it :)

File: store/mongo/migration_1_7_0.go (hunk 5 above)
Comment (merlin-northern) [reply store/mongo/migration_1_7_0.go:39]: same here! :)

File: store/datastore.go (hunk 4 above)
Comment (merlin-northern) [reply store/datastore.go]: sure.thanks. corrected.

File: cmd/commands_test.go (hunk 3 above)
Comment (merlin-northern) [reply cmd/commands_test.go]: sure thanks corrected

File: cmd/commands.go (hunk 2 above)
Comment (merlin-northern) [reply cmd/commands.go]: good catch! thanks! fixed

File: cmd/commands.go (hunk 1 above)
Comment (merlin-northern) [reply cmd/commands.go]: sure thanks corrected
